### PR TITLE
feature(ui-ux): implemented action button portfolio 2.0

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      })
+    }, {
+      name: translate('components/ActionButtons', 'Transactions'),
+      icon: 'calendar',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap')) {
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:53]: i think if there is no future swap, the button will not be display too, according to figma

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      name: translate('components/ActionButtons', 'Receive'),
+      icon: 'arrow-down-left',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('Receive')
+    }, {
+      name: translate('components/ActionButtons', 'Swap'),
+      icon: 'repeat',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate({
+        name: 'Convert',
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:41]: Should navigate to composite swap screen

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/PortfolioScreen.tsx [hunk 3]
```diff
@@ -477,6 +479,7 @@ export function PortfolioScreen ({ navigation }: Props): JSX.Element {
           denominationCurrency={denominationCurrency}
           setDenominationCurrency={setDenominationCurrency}
         />
+        <ActionButtons />
         <Announcements />
         <BalanceActionSection navigation={navigation} isZeroBalance={isZeroBalance} />
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/PortfolioScreen.tsx:484]: let's remove the old one

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap') && futureSwaps.length > 0) {
+    actions.splice(2, 0, {
+      name: translate('components/ActionButtons', 'Future swap'),
+      icon: 'clock',
+      iconType: 'Feather',
+      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length + 1,
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:63]: i think we dont need to +1 right?
```suggestion
      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length,
```

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/PortfolioScreen.tsx [hunk 5]
```diff
@@ -499,8 +500,8 @@ export function PortfolioScreen ({ navigation }: Props): JSX.Element {
           denominationCurrency={denominationCurrency}
           setDenominationCurrency={setDenominationCurrency}
         />
+        <ActionButtons />
         <Announcements />
-        <BalanceActionSection navigation={navigation} isZeroBalance={isZeroBalance} />
         {hasPendingFutureSwap && <FutureSwapCta navigation={navigation} />}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/PortfolioScreen.tsx:505]: we can remove this now

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { ScrollView, Text } from 'react-native'
+import { useSelector } from 'react-redux'
+import { PortfolioParamList } from '../PortfolioNavigator'
+
+export interface ActionButtonsProps {
+  name: string
+  icon: IconName
+  iconType: IconType
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:15]: add `iconSize` to customize size for icons

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        style={tailwind('rounded-full w-15 h-15 items-center justify-center mx-2.5')}
+        onPress={props.onPress}
+        testID={props.testID}
+      >
+        <ThemedIcon
+          dark={tailwind('text-mono-dark-v2-900')}
+          light={tailwind('text-mono-light-v2-900')}
+          iconType={props.iconType}
+          name={props.icon}
+          size={28}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:98]: add `style={tailwind('ml-px')}`

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    }, {
+      name: translate('components/ActionButtons', 'Transactions'),
+      icon: 'calendar',
+      testID: 'transaction_button',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap') && futureSwaps.length > 0) {
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx:62]: @fullstackninja864 I feel this one is more expensive. What if we just use if statements for rendering?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**